fda/magnitude_shape_plot.py

```diff
@@ -0,0 +1,527 @@
+                        [-0.5],
+                        [ 1. ],
+                        [ 1. ],
+                        [ 0.5]],
+        <BLANKLINE>
+                       [[-0.5],
+                        [-0.5],
+                        [-0.5],
+                        [-1. ],
+                        [-1. ],
+                        [-1. ]]]),
+                sample_points=[array([ 0,  2,  4,  6,  8, 10])],
+                domain_range=array([[ 0, 10]]),
+                dataset_label=None,
+                axes_labels=None,
+                extrapolation=None,
+                interpolator=GridSplineInterpolator(interpolation_order=1, smoothness_parameter=0.0, monotone=False),
+                keepdims=False),
+            depth_method=modified_band_depth,
+            dim_weights=None,
+            pointwise_weights=None,
+            alpha=0.993,
+            points=array([[ 1.66666667,  0.74074074],
+                   [ 0.        ,  0.        ],
+                   [-0.73333333,  0.36740741],
+                   [-1.        ,  0.53333333]]),
+            outliers=array([0, 0, 0, 0]),
+            colormap=seismic,
+            color=0.2,
+            outliercol=(0.8,),
+            xlabel='MO',
+            ylabel='VO',
+            title='MS-Plot')
+    """
+
+    def __init__(self, fdatagrid, depth_method=modified_band_depth,
+                 dim_weights=None,
+                 pointwise_weights=None, alpha=0.993):
+        """Initialization of the MagnitudeShapePlot class.
+
+        Args:
+            fdatagrid (FDataGrid): Object containing the data.
+            depth_method (:ref:`depth measure <depth-measures>`, optional): Method
+                used to order the data. Defaults to :func:`modified band depth
+                <fda.depth_measures.modified_band_depth>`.
+            dim_weights (array_like, optional): an array containing the weights
+                of each of the dimensions of the image.
+            pointwise_weights (array_like, optional): an array containing the
+                weights of each points of discretisati on where values have been
+                recorded.
+            alpha(float, optional): Denotes the quantile to choose the cutoff
+                value for detecting outliers Defaults to 0.993, which is used
+                in the classical boxplot.
+        """
+
+        if fdatagrid.ndim_image > 1:
+            raise NotImplementedError("Only support 1 dimension on the image.")
+
+        # The depths of the samples are calculated giving them an ordering.
+        _, mean_dir_outl, variation_dir_outl = directional_outlyingness(
+            fdatagrid,
+            depth_method,
+            dim_weights,
+            pointwise_weights)
+
+        points = np.array(list(zip(mean_dir_outl, variation_dir_outl))).astype(
+            float)
+
+        # The square mahalanobis distances of the samples are calulated using MCD.
+        cov = MinCovDet(store_precision=True).fit(points)
```

Comment:
Are points Gaussian?

Reply:
They are assumed to follow a normal distribution.

Reply:
Ok, then